google-cloud-storage/lib/google/cloud/storage/file.rb

```diff
@@ -775,6 +777,35 @@ def determine_issuer options = {}
               @service.credentials.issuer
           end
 
+          def post_object options
+            options = apply_option_defaults options
+
+            fields = {
+              key: ext_path.sub("/", "")
```

Comment:
I'm following what the example in the docs says:

```md
Key: 
The name of the object that you are uploading.
If you specify the object name without the bucket name, 
you must append the bucket name to the URI (for example, storage.googleapis.com/<bucket>). 
If you specify the bucket name and object name (for example, bucket/object),
you must specify only the URI (for example, storage.googleapis.com).
You can also use the ${filename} variable if a user is providing a file name.
Also, if you are using a bucket name that is a CNAME redirect,
be sure that the bucket name is specified only once in your request.
```

This way, the URL is `https://storage.googleapis.com` and the KEY is `mybucket/myobject`.

To be honest, I didn't try to leave the ext path as it was, but I know this version works 😄